src/activity_item_handling.cpp

```diff
@@ -1159,12 +1153,566 @@ void activity_on_turn_blueprint_move( player_activity &, player &p )
+    seed_map_t seed_names;
+    // check how many seeds we may require
+    for( const tripoint &src : src_sorted ) {
+        const zone_data *zone = mgr.get_zone_at( src );
+        if( !zone ) {
+            continue;
+        }
+        tripoint src_loc = g->m.getlocal( src );
+        if( g->m.has_flag_ter_or_furn( "PLANTABLE", src_loc ) && !g->m.has_items( src_loc ) &&
+            !g->m.has_furn( src_loc ) ) {
+            const auto options = dynamic_cast<const plot_options &>( zone->get_options() );
+            const std::string seed = options.get_seed();
+            std::pair<seed_map_t::iterator, bool> has_insert = seed_names.insert( std::make_pair( seed, 1 ) );
+            if( !has_insert.second ) {
+                seed_names[seed]++;
+            }
+        }
+    }
+    // what do we have in our inventory? compared to what we need?
+    std::vector<item *> seed_inv = p->items_with( []( const item & itm ) {
+        return itm.is_seed();
+    } );
+    for( const item *elem : seed_inv ) {
+        std::map<std::string, int>::iterator map_it;
+        map_it = seed_names.find( elem->typeId() );
+        if( map_it != seed_names.end() ) {
+            // if we have some in our inventory already, remove it from the wanted list.
+            seed_names.find( elem->typeId() )->second -= 1;
+            if( seed_names.find( elem->typeId() )->second <= 0 ) {
+                seed_names.erase( map_it );
+            }
+        }
+    }
+    // Nuke the current activity, leaving the backlog alone.
+    p->activity = player_activity();
+    // search in a radius to find the required seeds.
+    const std::unordered_set<tripoint> &src_set_loot_seeds = mgr.get_near(
+                zone_type_id( "LOOT_SEEDS" ), abspos, 60 );
+    const zone_type_id no_pickup( "NO_NPC_PICKUP" );
+
+    const std::vector<tripoint> &src_set_loot_seeds_sorted = get_sorted_tiles_by_distance( abspos,
+            src_set_loot_seeds );
+
+    bool found_any = false;
+    std::vector<tripoint> wanted_seed_points;
+    for( const tripoint &pabs : src_set_loot_seeds_sorted ) {
+        bool found_one = false;
+        tripoint p_loc = g->m.getlocal( pabs );
+        // dont pickup from this spot, even if its got what we need.
+        if( g->check_zone( no_pickup, p_loc ) ) {
+            continue;
+        }
+        // otherwise find the right seed
+        const map_stack items_at = g->m.i_at( p_loc );
+        for( const item &elem : items_at ) {
+            if( elem.is_seed() ) {
+                std::map<std::string, int>::iterator map_it;
+                map_it = seed_names.find( elem.typeId() );
+                if( map_it != seed_names.end() ) {
+                    found_any = true;
+                    found_one = true;
+                    break;
+                }
+            }
+        }
+        if( found_one ) {
+            wanted_seed_points.push_back( p_loc );
+        }
+    }
+    if( !found_any ) {
```

Comment:
```suggestion
    if( wanted_seed_points.empty() ) {
```
No need for a separate variable.